engine/src/main/java/org/entando/entando/web/common/model/RestListRequest.java

```diff
@@ -0,0 +1,165 @@
+/*
+ * Copyright 2015-Present Entando Inc. (http://www.entando.com) All rights reserved.
+ *
+ * This library is free software; you can redistribute it and/or modify it under
+ * the terms of the GNU Lesser General Public License as published by the Free
+ * Software Foundation; either version 2.1 of the License, or (at your option)
+ * any later version.
+ *
+ * This library is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
+ * details.
+ */
+package org.entando.entando.web.common.model;
+
+import java.util.Arrays;
+
+import com.agiletec.aps.system.common.FieldSearchFilter;
+import org.apache.commons.lang.StringEscapeUtils;
+import org.apache.commons.lang3.ArrayUtils;
+import org.apache.commons.lang3.StringUtils;
+
+public class RestListRequest {
+
+    private String sort;
+    private String direction;
+
+    private Integer pageNum = 0;
+    private Integer pageSize = 5;
+
+    private Filter[] filter;
+
+    public Filter[] getFilter() {
+        return filter;
+    }
+
+    public void setFilter(Filter[] filter) {
+        this.filter = filter;
+    }
+
+    public String getSort() {
+        return sort;
+    }
+
+    public void setSort(String sort) {
+        this.sort = sort;
+    }
+
+    public String getDirection() {
+        return direction;
+    }
+
+    public void setDirection(String direction) {
+        this.direction = direction;
+    }
+
+    public Integer getPageNum() {
+        return pageNum;
+    }
+
+    public void setPageNum(Integer pageNum) {
+        this.pageNum = pageNum;
+    }
+
+    public Integer getPageSize() {
+        return pageSize;
+    }
+
+    public void setPageSize(Integer pageSize) {
+        this.pageSize = pageSize;
+    }
+
+    public void addFilter(Filter filter) {
+        this.filter = ArrayUtils.add(this.filter, filter);
+    }
+
+    @SuppressWarnings("rawtypes")
+    public FieldSearchFilter[] getFieldSearchFilters() {
+        FieldSearchFilter[] filters = new FieldSearchFilter[0];
+        if (null != filter && filter.length > 0) {
```

Comment:
Going forward, lets make new code List<> and use streams. Then if we absolutely have to return an array we can convert it right before return.